KeryxConfig augmentation breaks framework config assignment in api.ts

## Problem

Follow-up to #223 / #224. The v0.11.1 fix changed `KeryxConfig` from a `type` to an `interface`, which allows module augmentation. However, when an app augments the interface:

```typescript
// app config/audit.ts
declare module "keryx" {
  interface KeryxConfig {
    audit: typeof configAudit;
  }
}
```

TypeScript errors on **keryx's own `api.ts:41`**:

```
api.ts(41,3): error TS2741: Property 'audit' is missing in type '{ actions: ...; channels: ...; ... }' but required in type 'KeryxConfig'.
```

This is because `api.ts` does:

```typescript
declare namespace globalThis {
  let config: KeryxConfig;  // ← typed as KeryxConfig
}
// ...
globalThis.config = Config;  // ← Config is the framework literal, missing 'audit'
```

The framework literal doesn't have the augmented properties yet (they're merged later by `deepMerge` in `API.loadConfig()`), so the assignment fails.

## Suggested Fix

Use a type assertion on the assignment in `api.ts`:

```typescript
globalThis.config = Config as KeryxConfig;
```

This is safe because `deepMerge` will add the app-level properties before any user code accesses them. The assertion bridges the gap between the initial framework-only config and the final merged config.

## Environment

- keryx 0.11.1
- Bun 1.3.x
- TypeScript 5.x

--- a/packages/keryx/api.ts
+++ b/packages/keryx/api.ts
@@ -38,7 +38,7 @@ if (!globalThis.api) {
   // @ts-ignore — augmented API properties (db, redis, etc.) are set later by initializers at runtime
   globalThis.api = new API();
   globalThis.logger = globalThis.api.logger;
-  globalThis.config = Config;
+  globalThis.config = Config as KeryxConfig;
 }
 
 /** The global API singleton. All framework state (actions, db, redis, etc.) is accessed through this object. */

--- a/packages/keryx/package.json
+++ b/packages/keryx/package.json
@@ -1,6 +1,6 @@
 {
   "name": "keryx",
-  "version": "0.11.1",
+  "version": "0.11.2",
   "module": "index.ts",
   "type": "module",
   "license": "MIT",

